studio.py

```diff
@@ -14,14 +22,211 @@
+        if not os.path.exists(self.path):
+            print("Creating recording path: %s" % self.path)
+            os.mkdir(self.path)
+
+        recordings = self.parse_sites_json()
+
+        for recording in recordings:
+            if not os.path.exists(os.path.dirname(recording["recording_path"])):
+                print(
+                        "Creating recording path: %s"
+                        % os.path.dirname(recording["recording_path"])
+                )
+                os.mkdir(os.path.dirname(recording["recording_path"]))
+
+            if not recording.get("login", False):
+                with PROXYS[self.proxy](
+                        path=recording["recording_path"], mode="record"
+                ) as proxy_service:
+                    app_service = APPS[self.app](
+                        proxy_service, self.certutil, self.app_path
+                    )
+                    app_service.start(recording["url"])
+
+                    if not recording.get("login", None):
+                        for i in range(0, RECORD_TIMEOUT):
+                            print(i)
+                            time.sleep(1)
+                    else:
+                        time.sleep(5)
+                        raw_input("Do user input and press <Return>")
+
+                    app_service.screen_shot(recording["screen_shot_path"])
+                    self.information["app_info"] = app_service.app_information()
+                    app_service.stop()
+
+            self.update_json_information(recording)
+            self.generate_zip_file(recording)
+            self.generate_manifest_file(recording)
+
+    def parse_sites_json(self):
+        print("Parsing sites json")
+        recordings = []
+        if self.sites is not None:
+            with open(self.sites, "r") as sites:
+                recordings_json = json.loads(sites.read())
+                self.information["app"] = self.app.lower()
+            if self.app in ["Firefox", "Chrome"]:
+                self.information["os_name"] = platform.system()
+                self.information["os_version"] = platform.release()
+                platform_name = platform.system().lower()
+
+            elif self.app in ["GeckoViewExample", "Fenix", "Fennec", "Refbrow"]:
+
+                device = ADBAndroid()
+                self.information["device_name"] = device.shell_output(
+                    "getprop ro.product.model"
+                )
+                self.information["os_name"] = device.shell_output(
+                    "getprop ro.build.user"
+                )
+                self.information["os_version"] = device.shell_output(
+                    "getprop ro.build.version.release"
+                )
+                if self.information["device_name"] == "Pixel 2":
+                    platform_name = "pixel2"
+                elif self.information["device_name"] == "Moto G (5)":
+                    platform_name = "motog5"
```

Comment:
To avoid having to maintain this list, let's strip anything that's not alphanumerical from ro.product.model and convert it to lower case.